src/logic/strategy.ts

```diff
@@ -101,7 +102,11 @@ class Strategy {
           fromBlock: this.startBlockNumber-1,
           filter: { batchId: bracketDepositEvents[0].batchId, user: bracketAddress }
         });
-  
+ 
+        // collect all withdrawRequest events, to determine if a strategy was withdraw requested yet
+        
+        // find blocks for all withdraw requests in order to determine time until claim is available
+        // todo: implement batchexchange time modulo to determine times more precisely
```

Comment:
These times are determined based on the batch id, which is a function of the current timestamp.

https://docs.gnosis.io/protocol/docs/intro-batches/#-how-can-i-get-the-batch-id-for-a-given-time

Withdraw request events have a batchId for when that withdraw is being requested at https://github.com/gnosis/dex-contracts/blob/master/contracts/EpochTokenLocker.sol#L39

A claim is available as soon as `current batchId` > `withdraw request batchId`

Although, there is an edge case, where you'll have to wait at least another batch https://github.com/gnosis/dex-react/blob/ac63a5cbd1710bab217ccee85e3f20acc2474fc7/src/components/DepositWidget/useRowActions.tsx#L171